spec/support/acts_as_tenant.rb

```diff
@@ -1,20 +1,44 @@
 # Test group helpers for setting the tenant for tests.
-module ActsAsTenant::TestGroupHelpers
-  # Sets the current tenant when running this group of tests.
-  #
-  # @param tenant [Symbol] The symbol containing the tenant to use for this group of
-  #                        tests. The tenant must have been set using a let construct.
-  # @param proc [Proc] The block containing the test definitions.
-  def with_tenant(tenant, &proc)
-    context "with tenant #{tenant.inspect}" do |*params|
-      before(:each) do
-        ActsAsTenant.current_tenant = send(tenant)
-        Capybara.app_host = 'http://' + send(tenant).host
+module ActsAsTenant
+  module TestGroupHelpers
+    # Sets the current tenant when running this group of tests.
+    #
+    # @param tenant [Symbol] The symbol containing the tenant to use for this group of
+    #                        tests. The tenant must have been set using a let construct.
+    # @param proc [Proc] The block containing the test definitions.
+    def with_tenant(tenant, &proc)
+      context "with tenant #{tenant.inspect}" do |*params|
+        before(:each) do
+          ActsAsTenant.current_tenant = send(tenant)
+        end
+        after(:each) do
+          ActsAsTenant.current_tenant = nil
+        end
+        instance_exec(*params, &proc)
       end
-      after(:each) do
-        ActsAsTenant.current_tenant = nil
+    end
+  end
+
+  module RequestTestHelpers
+    include TestGroupHelpers
+    # Sets the current tenant and host when running this group of tests.
+    #
+    # @param tenant [Symbol] The symbol containing the tenant to use for this group of
+    #                        tests. The tenant must have been set using a let construct.
+    # @param proc [Proc] The block containing the test definitions.
+    def with_tenant(tenant, &proc)
+      super(tenant) do
+        context 'with host set' do |*params|
```

Comment:
I'm not sure this is the best way to do it... the context is already set by `super`.